src/controller/AutoCommissioner.cpp

```diff
@@ -755,6 +770,19 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
             // storing the returned certs, so just return here without triggering the next stage.
             return NOCChainGenerated(report.Get<NocChain>().noc, report.Get<NocChain>().icac, report.Get<NocChain>().rcac,
                                      report.Get<NocChain>().ipk, report.Get<NocChain>().adminSubject);
+        case CommissioningStage::kICDSymmetricKeyGeneration: {
+            ICDRegistrationInfo registrationInfo = report.Get<ICDRegistrationInfo>();
+            mParams.SetICDCheckInNodeId(registrationInfo.checkInNodeId);
+            mParams.SetICDMonitoredSubject(registrationInfo.subjectId);
+            memcpy(mICDSymmetricKey, registrationInfo.key.data(), registrationInfo.key.size());
```

Comment:
This could use a comment about how registrationInfo.key.size() is a compile-time constant which matches the size of mICDSymmetricKey... Or even better a static_assert that the type of registrationInfo.key is a fixed span of size sizeof(mICDSymmetricKey) so it's clear all the sizes match and there are no buffer overruns on read or write.

Reply:
This logic is removed now.